src/batched/batchedmul.jl

```diff
@@ -42,6 +44,16 @@ This will be copied, as doing so is faster than `batched_mul_generic!`.
 Both this `copy` and `batched_mul_generic!` produce `@debug` messages,
 and setting for instance `ENV["JULIA_DEBUG"] = NNlib` will display them.
 """
+
+function batched_mul(x::AbstractArray{T1,N}, y::AbstractArray{T2,N}) where {T1,T2,N}
```

Comment:
My vote is to make this an internal `_batched_mul_4` or something for now. Partly because I think explaining what does and doesn't work becomes more complicated with this method. And that doesn't have to be solved to add attention.

Reply:
It's a pity to not make things available. Maybe I can leave the previous docstring unchanged and add a new one for the new method?